internal/config/config_test.go

```diff
@@ -122,30 +139,107 @@ func TestNormalizeFunc(t *testing.T) {
 	assert.Equal(t, expected, result)
 }
 
-func TestGetLog(t *testing.T) {
+func TestResolveLog(t *testing.T) {
 	viperInstance = viper.NewWithOptions(viper.KeyDelimiter(KeyDelimiter))
 	viperInstance.Set(LogLevelKey, "error")
 	viperInstance.Set(LogPathKey, "/var/log/test/test.log")
 
-	result := getLog()
+	result := resolveLog()
 	assert.Equal(t, "error", result.Level)
 	assert.Equal(t, "/var/log/test/test.log", result.Path)
 }
 
-func TestGetClient(t *testing.T) {
+func TestResolveClient(t *testing.T) {
 	viperInstance = viper.NewWithOptions(viper.KeyDelimiter(KeyDelimiter))
 	viperInstance.Set(ClientTimeoutKey, time.Hour)
 
-	result := getClient()
+	result := resolveClient()
 	assert.Equal(t, time.Hour, result.Timeout)
 }
 
-func TestGetAllowedDirectories(t *testing.T) {
-	viperInstance = viper.NewWithOptions(viper.KeyDelimiter(KeyDelimiter))
-	viperInstance.Set(ConfigDirectoriesKey, "/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules")
+func TestResolveCollector(t *testing.T) {
+	testDefault := getAgentConfig()
+	tests := []struct {
+		name      string
+		expected  *Collector
+		shouldErr bool
+		errMsg    string
+	}{
+		{
+			name:     "Test 1: Happy path",
+			expected: testDefault.Collector,
+		},
+		{
+			name: "Test 2: Non allowed path",
+			expected: &Collector{
+				ConfigPath: "/path/to/secret",
+			},
+			shouldErr: true,
+			errMsg:    "collector path /path/to/secret not allowed",
+		},
+		{
+			name: "Test 3: Unsupported Exporter",
+			expected: &Collector{
+				ConfigPath: testDefault.Collector.ConfigPath,
+				Exporters: []Exporter{
+					{
+						Type: "envoy",
+					},
+				},
+			},
+			shouldErr: true,
+			errMsg:    "unsupported exporter type: envoy",
+		},
+		{
+			name: "Test 4: Unsupported Processor",
+			expected: &Collector{
+				ConfigPath: testDefault.Collector.ConfigPath,
+				Exporters:  testDefault.Collector.Exporters,
+				Processors: []Processor{
+					{
+						Type: "custom-processor",
+					},
+				},
+			},
+			shouldErr: true,
+			errMsg:    "unsupported processor type: custom-processor",
+		},
+		{
+			name: "Test 5: Unsupported Receiver",
+			expected: &Collector{
+				ConfigPath: testDefault.Collector.ConfigPath,
+				Exporters:  testDefault.Collector.Exporters,
+				Receivers: []Receiver{
+					{
+						Type: "haproxy",
```

Comment:
maybe change this to nginx or a different receiver :)

Reply:
Renamed to the more serious "not-allowed"